#!/usr/bin/env python
PACKAGE = "grasp_utilities"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("forearm_height",    double_t,    0, "Altura en la que se encuentra el brazo respecto al plano", 0.2,  0.1, 0.6)
gen.add("forearm_angle", double_t, 0, "Angulo del brazo respecto al plano", 0.34, 0.0, 1.57)

gen.add("dim_x", double_t, 0, "Dimension del objeto (x)", 0.05, 0.01, 1.0)
gen.add("dim_y", double_t, 0, "Dimension del objeto (y)", 0.05, 0.01, 1.0)
gen.add("dim_z", double_t, 0, "Dimension del objeto (z)", 0.05, 0.01, 1.0)


gen.add("pos_x", double_t, 0, "Posicion del objeto respecto al eje en el plano(x)", 0.05, 0.01, 1.0)
gen.add("pos_y", double_t, 0, "Posicion del objeto respecto al eje en el plano(y)", 0.05, 0.01, 1.0)
gen.add("pos_z", double_t, 0, "Posicion del objeto respecto al eje en el plano(z)", 0.05, 0.01, 1.0)

type_enum = gen.enum([ gen.const("Cube",      int_t, 1, "Cube"),
                       gen.const("Sphere",     int_t, 2, "Sphere"),
                       gen.const("Cylinder",      int_t, 3, "Cylinder")],
                     "Enum for set type of object")

gen.add("form", int_t, 0, "Form of the object", 1, 1, 3, edit_method=type_enum)

exit(gen.generate(PACKAGE, "grasp_utilities", "object_marker_reconfiguration"))